Mathlib/Topology/Sets/OpenCover.lean

```diff
@@ -0,0 +1,54 @@
+/-
+Copyright (c) 2025 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+
+import Mathlib.Topology.Sets.Opens
+
+/-!
+# Open covers
+
+We define `IsOpenCover` as a predicate on indexed families of open sets in a topological space `X`,
+asserting that their union is `X`. This is an example of a declaration whose name is actually
+longer than its content; but giving it a name serves as a way of standardizing API.
+-/
+
+open Set Topology
+
+namespace TopologicalSpace
+
+/-- An indexed family of open sets whose union is `X`. -/
+def IsOpenCover {ι X : Type*} [TopologicalSpace X] (u : ι → Opens X) : Prop :=
+  iSup u = ⊤
```

Comment:
It's unclear to me whether we want this as a `Prop` on `u : ι → Opens X` or a structure on `u : ι → Set X`. I'm not saying the `Opens` version is bad, only opening the discussion.

If we keep the `Opens` version, I think it at least makes sense to have a constructor with `u : ι → Set X`.
```suggestion
def IsOpenCover {ι X : Type*} [TopologicalSpace X] (u : ι → Opens X) : Prop :=
  iSup u = ⊤
```

Reply:
I don't have a strong opinion on this one; I was mostly just following the existing code. What sort of syntax do you have in mind for a constructor? Maybe something like
```
lemma of_sets {v : ι → Set X} (h_open : ∀ i, IsOpen (v i)) (h_iUnion : ⋃ i, v i = univ) :
    IsOpenCover (fun i ↦ ⟨v i, h_open i⟩) := by
  simp [IsOpenCover, h_iUnion]
```